File: Dummy/src/Backend/src/Gateway/src/DomainUseCases/DummyItem/Actions/GetList/DummyItemGetListActionQuery.cs
namespace Makc2025.Dummy.Gateway.DomainUseCases.DummyItem.Actions.GetList;

/// <summary>
/// Запрос действия по получению списка фиктивных предметов.
/// </summary>
/// <param name="Page">Страница.</param>
/// <param name="Filter">Фильтр.</param>
public record DummyItemGetListActionQuery(
  QueryPage Page,
  DummyItemGetListActionQueryFilter Filter) : IQuery<Result<DummyItemListDTO>>;

File: Dummy/src/Frontend/next/Web/src/App/src/lib/DomainUseCases/DummyItem/Actions/GetList/DummyItemGetListActionQuery.ts
import { DummyItemGetListActionQueryFilter, QueryPage } from '@/lib';

export interface DummyItemGetListActionQuery {
  readonly filter?: DummyItemGetListActionQueryFilter | null;
  readonly page?: QueryPage | null;
}

export function createDummyItemGetListActionQuery(options?: Partial<DummyItemGetListActionQuery> | null): DummyItemGetListActionQuery {
  return {
    filter: options?.filter ?? null,
    page: options?.page ?? null,
  };
}